### PR TITLE
Board UI

### PR DESCRIPTION
**Pull request by PeterSzadurski**
### What this PR does (required):

- Adding Columns
- Adding Tasks

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/44846197/128436264-0ce53d5c-4a7e-46a5-b808-2f4c6a8e9ebd.mp4

### Any information needed to test this feature (required):

- Login / use demo account and modify the mock data

### Any issues with the current functionality (optional):

- css needs further refinement
- double click to edit title is inconsistent 
- add column color needs to match mockup
- add column icon needs to resize based off container thickness

### CODE REVIEW

File: client/src/components/Board/Board.tsx [hunk 1]
```diff
@@ -109,26 +126,115 @@ const Board: React.FunctionComponent = () => {
     }
   };
 
+  const addTask = (columnId: string) => {
+    const taskId = 'task-' + (Object.keys(state.mockData.tasks).length + 1);
+    const mockData = state.mockData;
+    mockData.tasks = {
+      ...mockData.tasks,
+      [taskId]: { name: 'Add title...', date: '', color: '#ffffff', id: taskId, isNew: true },
```
Comment (moffatethan) [inline client/src/components/Board/Board.tsx:134]: What is the plan with isNew?

File: client/src/components/Board/Column.tsx [hunk 2]
```diff
@@ -39,72 +48,87 @@ const Column: React.FunctionComponent<properties> = (props) => {
   const onMouseUp = () => {
     setVisable(-1);
   };
+  const addCard = () => {
+    console.log('Add card');
```
Comment (moffatethan) [inline client/src/components/Board/Column.tsx]: This can be removed please

File: client/src/components/Board/TaskModal.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+interface properties {
+  tasks: Tasks;
+  selectedTask: string;
+}
+
+const TaskModal: React.FunctionComponent<properties> = (props: properties) => {
+  const classes = useStyles();
+
+  if (props.tasks[props.selectedTask] == null) {
+    return <></>;
```
Comment (moffatethan) [inline client/src/components/Board/TaskModal.tsx]: We could just return null here couldn't we?

**Review by moffatethan — APPROVED** [review]
Left some comments for you to address, otherwise, good work Peter

File: client/src/components/Board/Board.tsx (hunk 1 above)
Comment (PeterSzadurski) [reply client/src/components/Board/Board.tsx:134]: Its used to check if a task should be editable & highlighted.

**Pull request merged**